File: Beecrowd_Parte1/BrincandoComPomekons_2185.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken()); // Quantidade de pilhas
        int M = Integer.parseInt(st.nextToken()); // Quantidade de alterações
        
        int[] pilhas = new int[N]; // Array para armazenar a quantidade de Pomekons em cada pilha
        
        // Lê as quantidades iniciais de Pomekons em cada pilha
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            pilhas[i] = Integer.parseInt(st.nextToken());
        }
        
        // Processa cada alteração
        for (int m = 0; m < M; m++) {
            st = new StringTokenizer(br.readLine());
            int X = Integer.parseInt(st.nextToken()) - 1; // 0-indexed
            int Y = Integer.parseInt(st.nextToken()) - 1; // 0-indexed
            int V = Integer.parseInt(st.nextToken());
            
            // Aplica a alteração nas pilhas
            for (int i = X; i <= Y; i++) {
                pilhas[i] = V;
            }
            
            // Verifica se Dabriel tem chance de ganhar após a alteração
            boolean possivel = false;
            for (int i = 0; i < N; i++) {
                if (pilhas[i] > 0) {
                    possivel = true;
                    break;
                }
            }
            
            // Imprime o resultado para a alteração atual
            if (possivel) {
                System.out.println("Possivel");
            } else {
                System.out.println("Impossivel");
            }
        }
        
        br.close();
    }
}

File: Beecrowd_Parte1/BrincandoComPomekons_2185.c
#include <stdio.h>

#define MAX_N 100005

int main() {
    int N, M;
    scanf("%d %d", &N, &M);

    // Array para armazenar as pilhas de Pomekons
    int pilhas[N + 1];
    for (int i = 1; i <= N; ++i) {
        scanf("%d", &pilhas[i]);
    }

    // Processamento das alterações nas pilhas
    for (int i = 0; i < M; ++i) {
        int X, Y, V;
        scanf("%d %d %d", &X, &Y, &V);
        
        // Aplicando a alteração diretamente nas pilhas
        for (int j = X; j <= Y; ++j) {
            pilhas[j] = V;
        }
    }

    // Verificação se Dabriel pode garantir a vitória
    int odd_count = 0;
    for (int i = 1; i <= N; ++i) {
        if (pilhas[i] % 2 != 0) {
            odd_count++;
        }
    }

    // Determinando se é possível para Dabriel ganhar sempre
    if (odd_count % 2 == 0) {
        printf("Possivel\n");
    } else {
        printf("Impossivel\n");
    }

    return 0;
}
